akka-stream/src/main/scala/akka/stream/scaladsl/DelayStrategy.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.scaladsl
+
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+
+/**
+ * Allows to manage delay and can be stateful to compute delay for any sequence of elements,
+ * all elements go through nextDelay() updating state and returning delay for each element
+ */
```

Comment:
Perhaps add something around concurrency here: "Instances are not shared among running streams, they can therefore safely be stateful"?